File: Depth-First Search/Binary tree/Searching/LET_no1448_Count Good Nodes in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        return helper(root, INT_MIN);
    }
    
    int helper(TreeNode* root, int maxNum) {
        if (!root)
            return 0;
        
        int res = 0;
        maxNum = max(maxNum, root->val);
        res += (root->val >= maxNum) + helper(root->left, maxNum) + helper(root->right, maxNum);
        
        return res;
    }
};

File: Depth-First Search/Binary tree/Searching/LET_no1448_Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        return self.dfs(root, float('-inf'))
    
    def dfs(self, root, maxNum):
        if not root:
            return 0
        
        maxNum = max(maxNum, root.val)
        res = 1 if root.val >= maxNum else 0
        res += self.dfs(root.left, maxNum) + self.dfs(root.right, maxNum)
        
        return res